apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -659,7 +659,16 @@ public function createShare(
 		$this->checkInheritedAttributes($share);
 
 		// Handle mail send
-		if ($sendMail === 'true' || $sendMail === 'false') {
+		if (is_null($sendMail)) {
+			// Define a default behavior when sendMail is not provided
+			if ($shareType === IShare::TYPE_EMAIL && strlen($shareWith) !== 0) {
```

Comment:
```suggestion
			if ($shareType === IShare::TYPE_EMAIL && $shareWith !== null && strlen($shareWith) > 0) {
```

$shareWith can be null as well and this would fail without a proper check.

Reply:
Actually no, `strlen(null) === 0`  is true.

Reply:
Hmmm, looks like this is deprecated on `php8.2+` though I tested it and it worked.

Reply:
Ok fine by me then, although it would be better to explicitly check it for the future.